python/cuml/cuml/tests/test_metrics.py

```diff
@@ -163,6 +163,8 @@ def test_r2_score(datatype, use_handle):
     np.testing.assert_almost_equal(score, 0.98, decimal=7)
 
 
+# Ignore FutureWarning: Using `__dataframe__` is deprecated
+@pytest.mark.filterwarnings("ignore::FutureWarning")
```

Comment:
AFAICT this test checks that `GridSearchCV` works with cudf, which is no longer true after `__dataframe__` was deprecated. IMO we should delete the test (or ask the cudf team to reconsider). Filtering the warning only puts things off until `__dataframe__` is removed when it'll just break again.

Reply:
Are you able to reproduce this warning locally? If so, can you please try commenting out the `__dataframe__` implementation in dataframe.py in cudf and try again? Is cuml using `__dataframe__` explicitly or is it an implicit path such that a different path would be taken if this implementation doesn't exist?

Reply:
That's a good point! From reading the sklearn code I _think_ sklearn has a path that would be taken once the `__dataframe__` path is fully removed, so maybe filtering out the warning for now is fine. I'll need to get a local cudf build setup to try, will do later today.

Reply:
Just realized `dataframe.py` isn't in cython, so this was an easy quick check. I can confirm that once the `__dataframe__` code is removed from cudf then things work again (though I can't say how efficiently). Using `filterwarnings` here seems fine (though with the recommendation to a more specific filter that Tim made above).

Reply:
The old `__dataframe__` code path wasn't actually "efficient" in any meaningful way. There is no device data transport happening, only metadata. That's actually the core problem with this protocol: it doesn't specify who is responsible for actually transferring data across device boundaries, leading to consumers having to make per-library distinctions. That's discussed a bit more in https://github.com/rapidsai/cudf/issues/17403.

Reply:
Thanks for testing so quickly, I was just spinning up my own dev environment to be able to verify this claim myself!

Reply:
Would prefer a more specific filter as well.

Reply:
I filed https://github.com/rapidsai/cuml/pull/6239 as a follow-up with the proposal above to make the filter more specific. My hope is that this PR passes CI and we can merge it as-is, then follow up with that PR.